app/controllers/samples_controller.rb

```diff
@@ -16,7 +16,7 @@ def index
     @samples = @samples.where("updated_at >= ?", params[:updated_at_from].to_time) unless params[:updated_at_from].blank?
     @samples = @samples.where("updated_at <= ?", params[:updated_at_to].to_time) unless params[:updated_at_to].blank?
     @samples = @samples.where("updated_at >= ?", params[:modified].to_time) unless params[:modified].blank?
-    @samples = @samples.order( params[:sort] + ' DESC') unless params[:sort].blank?
+    @samples = @samples.order(params[:sort] => :desc) unless params[:sort].blank?
```

Comment:
We should prevent injecting order by arbitrary columns in to the SQL query. While not necessarily a security risk (although it *can* leak some otherwise concealed information), it can create performance issues when huge tables are sorted by strange keys that are not meant for sorting.

I'd suggest adding a whitelist for column names that are available for sorting (for now it's probably just `updated_at`, right?).
It would also be a good idea to move this order mechanism into a scope on the `Sample` model.